### PR TITLE
Reset round timer upon instance start

### PR DESCRIPTION
**Pull request by amir-blox**
Added a call for round timeout upon instance start.

### CODE REVIEW

File: qbft/instance.go [hunk 1]
```diff
@@ -65,6 +65,8 @@ func (i *Instance) Start(value []byte, height Height) {
 			}
 		}
 
+		i.config.GetTimer().TimeoutForRound(FirstRound)
```
Comment (alonmuroch) [inline qbft/instance.go:68]: I wonder if this needs to come before the proposer if

File: qbft/spectest/tests/controller_spectest.go [hunk 2]
```diff
@@ -42,6 +42,11 @@ func (test *ControllerSpecTest) Run(t *testing.T) {
 		err := contr.StartNewInstance(runData.InputValue)
 		if err != nil {
 			lastErr = err
+		} else {
```
Comment (alonmuroch) [inline qbft/spectest/tests/controller_spectest.go:45]: what this suppose to test?

**Review by alonmuroch — CHANGES_REQUESTED** [review]

**Comment by alonmuroch** [comment]
rebase and squash

File: qbft/spectest/tests/controller_spectest.go (hunk 2 above)
Comment (amir-blox) [reply qbft/spectest/tests/controller_spectest.go:45]: That the timer was called upon start

**Comment by nivBlox** [comment]
@amir-blox no missing the timeout call to function?

**Comment by amir-blox** [comment]
@nivBlox the instance exposes `UponRoundTimeout()` which we should call from outside of spec.

**Comment by alonmuroch** [comment]
@amir-blox rebase  and squash

**Pull request merged**